src/test/java/co/rsk/bitcoinj/script/ScriptTest.java

```diff
@@ -472,84 +464,154 @@ public void numberBuilderNegative() {
+    }
+
     @Test
     public void getSigInsertionIndex_no_fast_bridge_redeem_script() {
-        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(btcECKeyList);
+        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(federationKeys);
         testGetSigInsertionIndex(redeemScript);
     }
 
+    @Test
+    public void getSigInsertionIndexWhenEmptyScriptShouldThrowScriptException() {
+        Script redeemScript = new ScriptBuilder().build();
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+
+        int sigInsertionIndex = redeemScript.getSigInsertionIndex(hashForSignature, signingKey);
+        Assert.assertEquals(0, sigInsertionIndex);
+    }
+
+    @Test
+    public void getSigInsertionIndexWhenP2shScriptShouldThrowScriptException() {
+        Script p2shOutputScript = ScriptBuilder.createP2SHOutputScript(
+            expectedNumberOfRequiredSignatures,
+            federationKeys
+        );
+
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+
+        int sigInsertionIndex = p2shOutputScript.getSigInsertionIndex(hashForSignature, signingKey);
+        Assert.assertEquals(0, sigInsertionIndex);
+    }
+
+    @Test
+    public void getSigInsertionIndexWhenTestnetHardcodeRedeemScriptThrowScriptException() {
+        final byte[] ERP_TESTNET_REDEEM_SCRIPT_BYTES = Utils.HEX.decode("6453210208f40073a9e43b3e9103acec79767a6de9b0409749884e989960fee578012fce210225e892391625854128c5c4ea4340de0c2a70570f33db53426fc9c746597a03f42102afc230c2d355b1a577682b07bc2646041b5d0177af0f98395a46018da699b6da210344a3c38cd59afcba3edcebe143e025574594b001700dec41e59409bdbd0f2a0921039a060badbeb24bee49eb2063f616c0f0f0765d4ca646b20a88ce828f259fcdb955670300cd50b27552210216c23b2ea8e4f11c3f9e22711addb1d16a93964796913830856b568cc3ea21d3210275562901dd8faae20de0a4166362a4f82188db77dbed4ca887422ea1ec185f1421034db69f2112f4fb1bb6141bf6e2bd6631f0484d0bd95b16767902c9fe219d4a6f5368ae");
+        Script erpTestnetRedeemScript = new Script(ERP_TESTNET_REDEEM_SCRIPT_BYTES);
+
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+
+        int sigInsertionIndex = erpTestnetRedeemScript.getSigInsertionIndex(hashForSignature, signingKey);
+        Assert.assertEquals(0, sigInsertionIndex);
+    }
+
+    @Test
+    public void getSigInsertionIndexWhenMalformedRedeemScriptThrowScriptException() {
+        Script customRedeemScript = new Script(new byte[2]);
+
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+
+        int sigInsertionIndex = customRedeemScript.getSigInsertionIndex(hashForSignature, signingKey);
+        Assert.assertEquals(0, sigInsertionIndex);
```

Comment:
Does it return 0 because of the script or because of the hashForSignature? I'd check that very carefully here and in the other tests

Reply:
It returns the default value, zero because it is identified as a no-redeem script.

Reply:
Added more assertion, checking that the redeemScriptParser got from the `redeemScriptParserFactory.get` is a `NoRedeemScript` instance, which returns zero for getSigInsertionIndex calls by default.